bdkwallet/manager.go

```diff
@@ -0,0 +1,111 @@
+package bdkwallet
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/utreexo/utreexod/blockchain"
+	"github.com/utreexo/utreexod/btcutil"
+	"github.com/utreexo/utreexod/chaincfg"
+	"github.com/utreexo/utreexod/mempool"
+)
+
+var defaultWalletPath = "bdkwallet"
+var defaultWalletFileName = "default.dat"
+
+// ManagerConfig is a configuration struct used to
+type ManagerConfig struct {
+	Chain       *blockchain.BlockChain
+	TxMemPool   *mempool.TxPool
+	ChainParams *chaincfg.Params
+	DataDir     string
+}
+
+type Manager struct {
+	config ManagerConfig
+	wallet Wallet // wallet does not need a mutex as it's done in Rust
```

Comment:
I was trying to get the rpc to work and I had to make the wallet public since the server is holding onto a `bdkwallet.Manager` and that didn't allow for access into the interface methods for the rpcserver.

Would this change be ok? Just making sure there isn't a specific reason as to the wallet being private